From b46d4de70f4b9b96ee5277037dd79ba2d7f75751 Mon Sep 17 00:00:00 2001
From: Fabian Pflug <f.pflug@pengutronix.de>
Date: Thu, 17 Apr 2025 10:35:16 +0200
Subject: [PATCH] ARM: dts: tqma93xx add dummy supply to regulator

The regulator for the usdhc2 references the BUCK4 as its input.
The BUCK4 is part of the pca9451, which in turn is not a regulator
in barebox, so there is a problem during initialisation:

WARNING: imx-esdhc 42860000.mmc@42860000.of: Failed to get 'vmmc' regulator (ignored)

Since the dependency could not be resolved, the reg_usdhc2_vmmc
regulator is not correctly initialized, leaving the gpio in a floating
state, which results in the GPIO being low, which in turn disables
the SD-Card.

Signed-off-by: Fabian Pflug <f.pflug@pengutronix.de>
---
 arch/arm/dts/imx93-tqma93xx.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/dts/imx93-tqma93xx.dtsi b/arch/arm/dts/imx93-tqma93xx.dtsi
index 40425e39a7..aafeadc275 100644
--- a/arch/arm/dts/imx93-tqma93xx.dtsi
+++ b/arch/arm/dts/imx93-tqma93xx.dtsi
@@ -15,6 +15,9 @@ pca9451a: pmic@25 {
 		reg = <0x25>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_pca9451>;
+		regulators {
+			barebox,allow-dummy-supply;
+		};
 	};
 };
 
-- 
2.39.5

